File: groups/group1/Collatz.java
import java.util.Scanner;

public class Collatz {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String numberText = scanner.hasNextLine() ? scanner.nextLine() : "";

        if (!hasValidInput(numberText)) {
            return;
        }

        int number = Integer.parseInt(numberText.trim());
        System.out.println("Collatz sequence for " + number + ":");

        while (number != 1) {
            System.out.print(number + " ");
            if (number % 2 == 0) {
                number = number / 2;
            } else {
                number = (number * 3) + 1;
            }
        }

        System.out.println(1);
    }

    static boolean hasValidInput(String numberText) {
        if (!isInteger(numberText)) {
            System.out.println("Invalid input. Enter a positive integer greater than 0.");
            return false;
        }

        int number = Integer.parseInt(numberText.trim());
        if (number <= 0) {
            System.out.println("Invalid input. Number must be greater than 0.");
            return false;
        }

        return true;
    }

    static boolean isInteger(String value) {
        try {
            Integer.parseInt(value.trim());
            return true;
        } catch (NumberFormatException error) {
            return false;
        }
    }
}

File: groups/group1/Collatz.js
function getCollatzSequence(n) {
  const sequence = [];
  let current = n;

  while (current !== 1) {
    sequence.push(current);
    current = current % 2 !== 0 ? (3 * current) + 1 : Math.floor(current / 2);
  }

  sequence.push(1);
  return sequence;
}

function parsePositiveOddInt(rawValue) {
  const trimmed = String(rawValue ?? "").trim();

  if (!trimmed) {
    throw new Error("Please enter a value before calculating.");
  }

  if (!/^-?\d+$/.test(trimmed)) {
    throw new Error("Invalid input — please enter a valid integer.");
  }

  const value = Number.parseInt(trimmed, 10);

  if (value <= 0 || value % 2 === 0) {
    throw new Error("Invalid input — please enter a positive odd integer.");
  }

  return value;
}

function main(inputs) {
  const startingNumber = parsePositiveOddInt(inputs[0]);
  const sequence = getCollatzSequence(startingNumber);
  const peak = Math.max(...sequence);

  return [
    `Collatz sequence for ${startingNumber}:`,
    sequence.join(", "),
    "",
    `Terms: ${sequence.length}`,
    `Steps: ${sequence.length - 1}`,
    `Peak value: ${peak}`,
    `Initial value: ${startingNumber}`,
  ].join("\n");
}
